src/containerapp/azext_containerapp/custom.py

```diff
@@ -5218,3 +5206,220 @@ def create_connected_environment(cmd,
+            return list(filter(both_match, r))
+        except Exception as e:
+            handle_raw_exception(e)
+
+
+def connected_env_upload_certificate(cmd, name, resource_group_name, certificate_file, certificate_name=None, certificate_password=None, location=None, prompt=False):
+    _validate_subscription_registered(cmd, CONTAINER_APPS_RP)
+
+    blob, thumbprint = load_cert_file(certificate_file, certificate_password)
+
+    cert_name = None
+    if certificate_name:
+        name_availability = connected_env_check_cert_name_availability(cmd, resource_group_name, name, certificate_name)
+        if not name_availability["nameAvailable"]:
+            if name_availability["reason"] == NAME_ALREADY_EXISTS:
+                msg = '{}. If continue with this name, it will be overwritten by the new certificate file.\nOverwrite?'
+                overwrite = True
+                if prompt:
+                    overwrite = prompt_y_n(msg.format(name_availability["message"]))
+                else:
+                    logger.warning('{}. It will be overwritten by the new certificate file.'.format(name_availability["message"]))
+                if overwrite:
+                    cert_name = certificate_name
+            else:
+                raise ValidationError(name_availability["message"])
+        else:
+            cert_name = certificate_name
+
+    while not cert_name:
+        random_name = generate_randomized_cert_name(thumbprint, name, resource_group_name)
+        check_result = connected_env_check_cert_name_availability(cmd, resource_group_name, name, random_name)
+        if check_result["nameAvailable"]:
+            cert_name = random_name
+        elif not check_result["nameAvailable"] and (check_result["reason"] == NAME_INVALID):
+            raise ValidationError(check_result["message"])
+
+    certificate = ContainerAppCertificateEnvelopeModel
+    certificate["properties"]["password"] = certificate_password
+    certificate["properties"]["value"] = blob
+    certificate["location"] = location
+    if not certificate["location"]:
+        try:
+            env = ConnectedEnvironmentClient.show(cmd, resource_group_name, name)
+            certificate["location"] = env["location"]
+        except Exception as e:
+            handle_raw_exception(e)
+
+    try:
+        r = ConnectedEnvironmentClient.create_or_update_certificate(cmd, resource_group_name, name, cert_name, certificate)
+        return r
+    except Exception as e:
+        handle_raw_exception(e)
+
+
+def connected_env_delete_certificate(cmd, resource_group_name, name, location=None, certificate=None, thumbprint=None):
+    _validate_subscription_registered(cmd, CONTAINER_APPS_RP)
+
+    if not certificate and not thumbprint:
+        raise RequiredArgumentMissingError('Please specify at least one of parameters: --certificate and --thumbprint')
+    certs = connected_env_list_certificates(cmd, name, resource_group_name, location, certificate, thumbprint)
```

Comment:
why do we need to list first?

Reply:
`az containerapp connected-env/env certificate remove` support to delete all certificates under the target environment, it will list certificates first, then delete them one by one.

Reply:
what does `-l` mean?